### PR TITLE
Make check for empty trusted proxies more strict

### PR DESCRIPTION
**Pull request by kesselb**
Follow-Up #14261

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Rebased to check if there are any CI failures.

**Comment by MorrisJobke** [comment]
@kesselb What is the status here?

**Comment by kesselb** [comment]
Looks good to me :rofl: $trustedProxies is always an array so this check would be more precise.

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

File: settings/Controller/CheckSetupController.php [hunk 1]
```diff
@@ -289,7 +289,7 @@ private function forwardedForHeadersWorking() {
 		$trustedProxies = $this->config->getSystemValue('trusted_proxies', []);
 		$remoteAddress = $this->request->getHeader('REMOTE_ADDR');
 
-		if (empty($trustedProxies) && $this->request->getHeader('X-Forwarded-Host')) {
+		if ($trustedProxies === [] && $this->request->getHeader('X-Forwarded-Host') !== '') {
```
Comment (nickvergessen) [inline settings/Controller/CheckSetupController.php]: `'trusted_proxies' => false,` (although misconfigured) did block this before, now it is accepted. Not sure if that is good/intended

Comment (MorrisJobke) [reply settings/Controller/CheckSetupController.php]: Then something like `is_array($trustedProxies)` makes the check more clear.

Comment (nickvergessen) [reply settings/Controller/CheckSetupController.php]: Which is below and makes exactly that flip 🙈

Comment (kesselb) [reply settings/Controller/CheckSetupController.php]: Show a warning if X-Forwarded-Host is present and $trustedProxies is empty or false sounds good to me.

**Comment by MorrisJobke** [comment]
> * tests/acceptance/features/apps.feature:43

"Fixed" with #14774 (caused by #14578)

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**